### ISSUE TITLE
Hang in Scripting.Classes.Inheritance.Tests.ps1

### ISSUE THREAD

**Issue by anmenaga** [PowerShell/PowerShell#3670]
This happened during the night build on Linux, resulting in a failed test run:

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> The build has been terminated

Can manually repro on `GitCommitId: v6.0.0-alpha.18` and `GitCommitId: v6.0.0-alpha.18-36-ga2268ab3ecb6c1addffbedba0606aa6c95809dd4-dirty`.

It feels like there is some sort of race - sometimes hang happens, sometimes - not.
Usually it happens within first 10 iterations of the test file.
There might be a product bug behind this.

Steps to reproduce
------------------
`1..50 | %{Invoke-Pester ./test/powershell/Language/Classes/scripting.Classes.inheritance.tests.ps1} `

Expected behavior
-----------------
100% pass

Actual behavior
---------------
PS hangs.

Example 1:
![ss1](https://cloud.githubusercontent.com/assets/11860095/25588692/36a3cd56-2e5e-11e7-97ed-3d6fb5f59b68.png)

Example 2:
![ss2](https://cloud.githubusercontent.com/assets/11860095/25588693/36a9ba86-2e5e-11e7-8f28-4f464047e4ec.png)

Environment data
----------------
```
Name                           Value                                           
----                           -----                                           
PSVersion                      6.0.0-alpha                                     
PSEdition                      Core                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                         
BuildVersion                   3.0.0.0                                         
GitCommitId                    v6.0.0-alpha.18                                 
CLRVersion                                                                     
WSManStackVersion              3.0                                             
PSRemotingProtocolVersion      2.3                                             
SerializationVersion           1.1.0.1   
```
```
uname -a
Linux anmenaga-d1 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1+deb8u2 (2017-03-07) x86_64 GNU/Linux
```
```
PS /home> Get-Content /etc/os-release -Raw                                                                  
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by adityapatwardhan** [comment]
It could be because of this https://github.com/dotnet/coreclr/issues/9745.
Workaround for another hung test here:  https://github.com/PowerShell/PowerShell/pull/3191

**Comment by iSazonov** [comment]
dotnet/coreclr#9745 milstone was changed to "Future" 😕 @daxian-dbw Isn't this a problem for the PowerShell 6.0 RTM release?

**Comment by daxian-dbw** [comment]
@iSazonov thanks for pointing out, I didn't know they moved it to future. I think it will be a potential problem to powershell RTM since our release will be crossgen'ed. 
cc/ @SteveL-MSFT @joeyaiello

**Comment by daxian-dbw** [comment]
@SteveL-MSFT @joeyaiello what is the rough time for powershell RTM? Are we following the same schedule of .NET Core 2.0 RTM? Either way, we need to talk to CoreCLR team to make sure the corssgen bug gets fixed before we go RTM.

**Comment by iSazonov** [comment]
@SteveL-MSFT  What is a conclusion here? Do we need a fix before RTM?

**Comment by SteveL-MSFT** [comment]
I'll see if coreclr can fix this for 2.0.0 servicing, otherwise, it'll be a documented `Known Issues` item

**Comment by daxian-dbw** [comment]
dotnet/coreclr#9745 has been fixed in .NET Core 2.0.4. PowerShell has moved to 2.0.4. We removed all workaround on our side and the hang doesn't happen again.
